mne/channels/_dig_montage_utils.py

```diff
@@ -188,27 +188,30 @@ def _foo_get_data_from_dig(dig):
 
 
 def _get_fid_coords(dig):
-    fid_coords = dict()
-    fid_coord_frames = []
+    fid_coords = Bunch(nasion=None, lpa=None, rpa=None)
+    fid_coord_frames = dict()
 
     for d in dig:
         if d['kind'] == FIFF.FIFFV_POINT_CARDINAL:
-            fid_coords[_cardinal_ident_mapping[d['ident']]] = d['r']
-            fid_coord_frames.append(d['coord_frame'])
+            key = _cardinal_ident_mapping[d['ident']]
+            fid_coords[key] = d['r']
+            fid_coord_frames[key] = d['coord_frame']
 
-    fid_coord_frames = set(fid_coord_frames)
+    coord_frame = None
+    if len(fid_coord_frames) > 0:
+        fid_coord_frames = set(fid_coord_frames.values())
 
-    if len(fid_coord_frames) != 1:
-        raise ValueError(
-            'All fiducial points must be in the same coordinate system '
-            '(got %s)' % len(fid_coord_frames)
-        )
+        if len(fid_coord_frames) > 1:
+            raise ValueError(
+                'All fiducial points must be in the same coordinate system '
+                '(got %s)' % len(fid_coord_frames)
+            )
 
-    coord_frame = fid_coord_frames.pop()
+        coord_frame = fid_coord_frames.pop()
 
-    if set(fid_coords.keys()) != set(_cardinal_ident_mapping.values()):
-        raise ValueError("Some fiducial points are missing (got %s)." %
-                         fid_coords.keys())
+        if set(fid_coords.keys()) != set(_cardinal_ident_mapping.values()):
```

Comment:
@agramfort this will always be true. 'cos `fid_coords = {'nasion': None, 'lpa': None ..}`

this should be tested against `fid_coord_frames` but before checking the set.


But I like how you simplified `__iadd__`.